### PR TITLE
Allow playback to start before all the album cover arts are loaded

### PR DESCRIPTION
**Pull request by paulijar**
After the performance fixes made in PR #517, there was still one problem in the perceived performance: When loading the Music app, all the cover images had to be completely loaded before playing the music was possible. This was kind of bad since playing music is the primary function of the application and seeing the cover images is more or less a nice-to-have feature. 

This PR addresses the issue by running the HTTP GET requests to retrieve the cover images via a queue. This way, browser can serve other requests (like downloading the music file) between getting the individual cover images.

The queued requests are not run in purely sequential order because this would increase the total time taken to download the cover images. Instead, at maximum 5 simultaneous requests are allowed. With this setup, there is no measurable difference in the total time taken to download the images but music playback can still be started with rather short delay. To easily implement this kind of "semi-sequential" behavior, the library ngQueue was added to the project.

### CODE REVIEW

File: js/app/directives/albumart.js [hunk 1]
```diff
@@ -19,7 +19,13 @@
...
  */
 
-angular.module('Music').directive('albumart', ['$http', function($http) {
+angular.module('Music').directive('albumart', ['$http', '$queueFactory', function($http, $queueFactory) {
+	// Calling $http.get immediately for all album cover images would be bad idea because it would
+	// block the playback until all the covers are loaded. Hence, we use queue which allows 5 simultaneous
+	// HTTP GET queries. This is faster than running them only one at a time, but still enables starting
+	// the playback with rather short delay.
+	var httpQueue = $queueFactory(5);
+
```
Comment (gnarula) [inline js/app/directives/albumart.js]: trailing whitespace

File: .gitignore [hunk 2]
```diff
@@ -55,6 +55,6 @@ coverage-html-integration
 # L10n
 l10n/templates/music.pot
 
-vendor/
 bin/
 tests/behat.yml
+/vendor/
```
Comment (gnarula) [inline .gitignore]: please use unix style line endings (LF) https://doc.owncloud.org/server/9.1/developer_manual/general/codingguidelines.html#coding

File: js/public/app.js [hunk 3]
```diff
@@ -498,7 +499,13 @@ angular.module('Music').controller('PlaylistController',
...
 }]);
 
-angular.module('Music').directive('albumart', ['$http', function($http) {
+angular.module('Music').directive('albumart', ['$http', '$queueFactory', function($http, $queueFactory) {
+	// Calling $http.get immediately for all album cover images would be bad idea because it would
+	// block the playback until all the covers are loaded. Hence, we use queue which allows 5 simultaneous
+	// HTTP GET queries. This is faster than running them only one at a time, but still enables starting
+	// the playback with rather short delay.
+	var httpQueue = $queueFactory(5);
+
```
Comment (gnarula) [inline js/public/app.js]: trailing whitespace

File: js/public/app.js [hunk 4]
```diff
@@ -498,7 +499,13 @@ angular.module('Music').controller('PlaylistController',
 
 }]);
 
-angular.module('Music').directive('albumart', ['$http', function($http) {
+angular.module('Music').directive('albumart', ['$http', '$queueFactory', function($http, $queueFactory) {
```
Comment (gnarula) [inline js/public/app.js]: please use LF line endings

**Comment by gnarula** [comment]
Looks good, thanks for your PR! Please make the changes I've mentioned in the comments and it's good to commit :)

**Comment by paulijar** [comment]
Thanks for your review, @gnarula. The findings are now fixed.

**Comment by gnarula** [comment]
Thanks! Merged :)

**Pull request merged**